mod_pbxproj/mod_pbxproj.py

```diff
@@ -1437,34 +1504,53 @@ def _printNewXCodeFormat(self, out, root, deep, enters=True, sort=False):
             out.write('}')
 
         elif isinstance(root, UserList):
-            out.write('(')
+            needs_indent = root_key != 'PRODUCT_BUNDLE_IDENTIFIER'
+            if needs_indent:
+                out.write('(')
 
-            if enters:
-                out.write('\n')
+                if enters:
+                    out.write('\n')
 
             for value in root:
-                if enters:
+                if enters and needs_indent:
                     out.write('\t' + deep)
 
-                self._printNewXCodeFormat(out, value, '\t' + deep, enters=enters)
-                out.write(',')
+                self._printNewXCodeFormat(out, value, None, '\t' + deep, enters=enters)
+                if needs_indent:
+                    out.write(',')
 
-                if enters:
+                if enters and needs_indent:
                     out.write('\n')
 
-            if enters:
-                out.write(deep)
+            if needs_indent:
+                if enters:
+                    out.write(deep)
 
-            out.write(')')
+                out.write(')')
 
         else:
-            if len(root) > 0 and re.match(regex, root).group(0) == root:
-                out.write(root.encode("utf-8"))
+            if len(root) > 0 and _escapeValuePat.match(root).group(0) == root:
+                out.write(_write_encode(root))
             else:
-                out.write('"' + XcodeProject.addslashes(root.encode("utf-8")) + '"')
+                out.write('"' + XcodeProject.addslashes(_write_encode(root)) + '"')
 
             if root in self.uuids:
-                out.write(" /* " + self.uuids[root].encode("utf-8") + " */")
+                name = self.uuids[root]
+
+                if root_key not in ['remoteGlobalIDString', 'TestTargetID']:
```

Comment:
Unity3d relies on comments to locate insertion points of elements into the project. Removing comments seems like a potential breaking change. I would advise to test this into with unity or omit the conditional part an always output the comment.